engine/src/main/java/io/zeebe/engine/processing/incident/CreateIncidentProcessor.java

```diff
@@ -91,12 +89,20 @@ private boolean tryRejectJobIncident(
   /** @return true if rejected, otherwise false */
   private boolean tryRejectWorkflowInstanceIncident(
       final long elementInstanceKey, final CommandControl<IncidentRecord> commandControl) {
-    final IndexedRecord failedRecord = elementInstanceState.getFailedRecord(elementInstanceKey);
-    final boolean noFailedRecord = failedRecord == null;
-    if (noFailedRecord) {
-      commandControl.reject(
-          RejectionType.NOT_FOUND, String.format(NO_FAILED_RECORD_MESSAGE, elementInstanceKey));
+    final var elementInstance = elementInstanceState.getInstance(elementInstanceKey);
+    if (elementInstance == null) {
+      // todo (@korthout): create an issue to enable this check
+      // this fails MultiInstanceIncidentTest shouldCreateIncidentIfOutputElementExpre...
+      // because an incident raised in the container processor does not interrupt the processing of
+      // the child element's complete and results in deleting the child element instance (where the
+      // incident relates to). So it is also impossible to resolve this incident.
+      //      commandControl.reject(
+      //          RejectionType.NOT_FOUND,
+      //          String.format(
+      //          NO_ELEMENT_INSTANCE_MESSAGE,
+      //          elementInstanceKey
+      //      ));
```

Comment:
I'll create an issue before merging this

Reply:
@Zelldon, I read this comment again, ready to create the issue, but now I'm hesitant. 

This situation will be solved when stop processing `Incident:Create` altogether, since the incident can then be created on the processing of the command that actually wants to write the incident (already creating the incident before the child element is completed and removed from state).

However, that means that when we merge this this bug exists. Should I maybe remove processing of the Incident:Create in this PR?

Reply:
How big will this change be?

Reply:
I don't think its big, but I think it is dependent on having all other element processors (that can raise incidents) migrated. Now this PR already fixes a known replay bug, so perhaps we should just postpone this fix to later

Reply:
Created https://github.com/camunda-cloud/zeebe/issues/6546 (and removed this from the pr, because it in the future we wont reject this command anymore anyways because with #6528 we won't have a create command anymore)